doc/Container Hardening/SONiC_container_hardening_HLD.md

```diff
@@ -0,0 +1,349 @@
+
+        $(DOCKER_DATABASE)_RUN_OPT += -t –-cap-add NET_ADMIN  # Changed by removing the --privileged flag and adding --cap-add flag
+
+### 4.2 net=host optimization
+
+Here we will provide a detailed example of how to switch from the `--net=host` configuration (host network) to the `--net=bridge` configuration paired with port forwarding in a specific container. We are using the database container as an example for this item.
+
+The original docker creation should be like in the example below:
+
+docker with host sharing:
+
+	docker create --privileged -t -v /etc/sonic:/etc/sonic:ro  \
+					--net=$NET \
+					-e RUNTIME_OWNER=local \
+					--uts=host \
+					--log-opt max-size=2M --log-opt max-file=5 \
+					--tmpfs /tmp \
+					$DB_OPT \
+					$REDIS_MNT \
+					-v /usr/share/sonic/device/$PLATFORM:/usr/share/sonic/platform:ro \
+					--tmpfs /var/tmp \
+					--env "NAMESPACE_ID"="$DEV" \
+					--env "NAMESPACE_PREFIX"="$NAMESPACE_PREFIX" \
+					--env "NAMESPACE_COUNT"=$NUM_ASIC \
+					--name=database_no_net \
+					--cap-drop=NET_ADMIN \
+					docker-database:latest
+
+To disable the sharing of the networking stack between the host and a container we need to remove the flag: `--net=host`. Because we have not specified any `--network` flag, the containers connect to the default bridge network `--net=bridge`.
+To support port forwarding we are required to add the flag:  `-p <port>:<port>`
+
+The "new" docker creation file database.sh can be seen in the code block below:
+
+Docker with port forwarding and default bridge network
+
+	docker create --privileged -t -v /etc/sonic:/etc/sonic:ro  \
+					**-p 6379:6379** \
+					-e RUNTIME_OWNER=local \
+					--uts=host \
+					--log-opt max-size=2M --log-opt max-file=5 \
+					--tmpfs /tmp \
+					$DB_OPT \
+					$REDIS_MNT \
+					-v /usr/share/sonic/device/$PLATFORM:/usr/share/sonic/platform:ro \
+					--tmpfs /var/tmp \
+					--env "NAMESPACE_ID"="$DEV" \
+					--env "NAMESPACE_PREFIX"="$NAMESPACE_PREFIX" \
+					--env "NAMESPACE_COUNT"=$NUM_ASIC \
+					--name=$DOCKERNAME \
+					docker-database:latest \
+
+**How we did it?**
+
+To create a docker with the flags above it is required to set the "new" flag in the file docker_image_ctl.js. Follow the call `docker create {{docker_image_run_opt}} \`: 
+and replace the `–--net=$NET`.
+docker flag generation
+
+	{%- if docker_container_name != "database" %}
+					--net=$NET \
+	{%- endif %}
+	{%- if docker_container_name == "database" %}
+					-p 6379:6379 \
+	{%- endif %}
+
+#### How to check?
+
+Go into the docker - `docker exec -it docker bash`
+Run `ifconfig`.
+
+On a docker with host network - you'll be able to view all physical interfaces.
+On a docker without host network - we'll see only eth0 and lo.
+
+## 5. SAI API
+
+N/A
+
+## 6. Configuration and management
+
+N/A - no configuration management/changes are required.
+
+### 6.1. Manifest (if the feature is an Application Extension)
+
+N/A
+
+### 6.2. CLI/YANG model Enhancements
+
```

Comment:
Is there a tool that knows what containers are running privileged mode?

Reply:
There might be, requires a bit of research.

Reply:
maybe you can check by `docker inspect container_name | grep Privileged`?

Reply:
This might be applicable for the --privileged flag, but this HLD is also discussing host network.
So while this might work for one of the HLD item (removing containers from privilege mode) it does not scope the whole feature list.
So this command might be a good start, but we should look at a more encompassing approach to check this (and not stop here).